File: 2016/01/1.py
P = (0, 0)
O = (1, 0)
for i in open("input").read().split(", "):
    if i[0] == 'L':
        O = (O[1], -O[0])
    elif i[0] == 'R':
        O = (-O[1], O[0])
    P = (P[0] + int(i[1:]) * O[0],
         P[1] + int(i[1:]) * O[1])
print(abs(P[0]) + abs(P[1]))

File: 2024/11/1.jl
function blink!(s)
    i = 1
    for i in eachindex(s)
        si = string(s[i])
        ds = length(si)
        if s[i] == 0
            s[i] = 1
        elseif iseven(ds)
            s[i] = parse(Int, si[1:ds÷2])
            push!(s, parse(Int, si[ds÷2+1:end]))
        else
            s[i] *= 2024
        end
        i += 1
    end
    return s
end

Base.:^(f, n) = n==0 ? identity : f ∘ f ^ (n-1)

i = parse.(Int, split(read("input", String)))
println(length((blink!^25)(i)))
